fastlane/lib/fastlane/actions/register_device.rb

```diff
@@ -19,8 +17,13 @@ def self.run(params)
         Spaceship.login(credentials.user, credentials.password)
         Spaceship.select_team
 
-        UI.user_error!("Passed invalid UDID: #{udid} for device: #{name}") unless UDID_REGEXP =~ udid
-        Spaceship::Device.create!(name: name, udid: udid)
+        begin
+          Spaceship::Device.create!(name: name, udid: udid)
+        rescue => ex
+          UI.error("Failed to register new device")
+          UI.error(ex.to_s)
+          return nil
```

Comment:
Same here...

I think we might want to make this...
```rb
UI.error(ex.to_s)
UI.crash!("Failed to register new device (name #{name}, UDID: #{udid})")
```

We want to make sure that we end execution and that an error return code gets bubbled to the top of this execution